### PR TITLE
Peterbranch

### PR DESCRIPTION
**Pull request by bagduch**
add address check method.

### CODE REVIEW

**Comment by djpnewton** [comment]
some files should not be added: wallet_eth.json, gethtxscan.log, *.db etc

File: xchwallet/btcwallet.cs [hunk 1]
```diff
@@ -66,6 +66,11 @@ public bool IsMainnet()
             return client.Network == Network.Main;
         }
 
+        public bool ContainsAddress(string address)
+        {
+                return true;
```
Comment (djpnewton) [inline xchwallet/btcwallet.cs]: raise an exception with "not yet implemented"

File: xchwallet/ethwallet.cs [hunk 2]
```diff
@@ -101,6 +101,26 @@ public bool IsMainnet()
...
+            // create a flag default false
+            bool flag = false;
+             if (wd.Accounts.Count>0)
+             {  
+                foreach(var item in wd.Accounts)
+                {
+                    foreach (var acct in item.Value)
+                       if(acct.Address==address)
+                       {
+                            flag= true;
```
Comment (djpnewton) [inline xchwallet/ethwallet.cs]: get rid of flag and just return true here

**Comment by bagduch** [comment]
update flag to ture or false

**Comment by bagduch** [comment]
Add the method for check address for BTC

File: xchwallet/btcwallet.cs [hunk 3]
```diff
@@ -66,6 +66,23 @@ public bool IsMainnet()
             return client.Network == Network.Main;
         }
 
+        public bool ContainsAddress(string address)
+        {
+            if(wd.Addresses.Count>0)
+            {
+                    foreach(var item in wd.Accounts)
```
Comment (djpnewton) [inline xchwallet/btcwallet.cs]: fix whitespace

File: test/wallet_eth.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (djpnewton) [inline test/wallet_eth.json]: remove this file

File: test/Program.cs [hunk 5]
```diff
@@ -21,10 +21,10 @@ static void Main(string[] args)
             */
 
             var filename = Path.Combine(Directory.GetCurrentDirectory(), "wallet_eth.json");
```
Comment (djpnewton) [inline test/Program.cs]: remove these changes

**Comment by djpnewton** [comment]
remove gethtxscan/gethtxscan.log

File: test/wallet_eth.json (hunk 4 above)
Comment (bagduch) [reply test/wallet_eth.json]: file removed

File: test/Program.cs (hunk 5 above)
Comment (bagduch) [reply test/Program.cs]: path removed

File: test/Program.cs [hunk 6]
```diff
@@ -20,11 +20,11 @@ static void Main(string[] args)
             wallet.Save(filename);
             */
 
-            var filename = Path.Combine(Directory.GetCurrentDirectory(), "wallet_eth.json");
-            var wallet = new EthWallet("12345678901234567890123456789012", filename, false, "https://ropsten.infura.io", "http://localhost:5001");
-            //Console.WriteLine(wallet.NewAddress("blah"));
-            //Console.WriteLine(wallet.NewAddress("blah2"));
-            var txs = wallet.GetTransactions("0x64E43D4f6023b6c8a644D06Ff60765F9C0ceF2Ec");
+
```
Comment (djpnewton) [inline test/Program.cs]: remove changes to this file

File: xchwallet/btcwallet.cs [hunk 7]
```diff
@@ -66,6 +66,22 @@ public bool IsMainnet()
             return client.Network == Network.Main;
         }
 
+        public bool ContainsAddress(string address)
+        {
+            if(wd.Addresses.Count>0)
```
Comment (djpnewton) [inline xchwallet/btcwallet.cs]: this check is not needed

File: .gitignore [hunk 8]
```diff
@@ -4,3 +4,4 @@
 
 obj/
 bin/
+test/Program.cs
```
Comment (djpnewton) [inline .gitignore]: revert this

File: test/Program.cs (hunk 6 above)
Comment (djpnewton) [reply test/Program.cs]: ...

**Comment by djpnewton** [comment]
remove gethtxscan/gethtxscan_testnet.db

File: xchwallet/btcwallet.cs [hunk 9]
```diff
@@ -66,6 +66,20 @@ public bool IsMainnet()
             return client.Network == Network.Main;
         }
 
+        public bool ContainsAddress(string address)
+        {
+
```
Comment (djpnewton) [inline xchwallet/btcwallet.cs]: fix whitespace!

**Pull request merged**